cmd/setup/51/01-permitted_orgs_function.sql

```diff
@@ -0,0 +1,77 @@
+DROP FUNCTION IF EXISTS eventstore.permitted_orgs;
+
+CREATE OR REPLACE FUNCTION eventstore.permitted_orgs(
+    instanceId TEXT
+    , userId TEXT
+    , perm TEXT
+    , system_user_roles TEXT[]
+    , filter_orgs TEXT
+
+    , org_ids OUT TEXT[]
+)
+	LANGUAGE 'plpgsql'
+	STABLE
+AS $$
+DECLARE
+	matched_roles TEXT[]; -- roles containing permission
+BEGIN
+	SELECT array_agg(rp.role) INTO matched_roles
+	FROM eventstore.role_permissions rp
+	WHERE rp.instance_id = instanceId
+	AND rp.permission = perm;
+
+  IF system_user_roles IS NOT NULL THEN
+    DECLARE
+      permission_found_in_system_roles bool;
+    BEGIN
+      SELECT result.role_found INTO permission_found_in_system_roles
+      FROM (SELECT matched_roles && system_user_roles AS role_found) AS result;
```

Comment:
For the system user we should only use the static role-permission definition from the runtime configuration. When using the instance level role-permissions, it is will be [possible in the future](https://github.com/zitadel/zitadel/issues/9189) to change the definition of the roles and effectively be able to lockout system users.

From the linked issue's acceptance criteria:

> Optional arguments are passed to the permission functions with the system user memberships and role permission mappings.